x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/IndicesPermission.java

```diff
@@ -609,6 +609,7 @@ public static class Group {
         private final String[] indices;
         private final StringMatcher indexNameMatcher;
         private final Supplier<Automaton> indexNameAutomaton;
+        // TODO: Use FieldPermissionsDefinition instead of FieldPermissions. The former is a better counterpart to query
         private final FieldPermissions fieldPermissions;
         private final Set<BytesReference> query;
```

Comment:
The TODO here is to note down that using `FieldPermissions` here is inconsistent when comparing to the `query` field (`Set<BytesReference>` instead of `DocumentPermissions`). At this level, it is better to use `FieldPermissionsDefinition` and only build `FieldPermissions` when computing `IndicesAccessControl`.